### DISCUSSION TITLE
Add batch inference

### DISCUSSION THREAD

**Discussion by egoriyaa** [amazon-science/chronos-forecasting#233] [Ideas]
Hi!
As I correctly understand, if dataset contains a lot of time series and inference is doing on gpu, model can fail due to OOM.
May be it will be better to pass "batch" of time series to model.

**Comment by lostella** [comment]
@egoriyaa of course, you can chunk your inference dataset in batches, and pass them to the model one batch at a time. We don't do that in the package since it's something that users can easily do around the `.predict` call in the way that is most convenient for them.

Autogluon (see [tutorial](https://auto.gluon.ai/stable/tutorials/timeseries/forecasting-chronos.html)) does this out of the box, if you're interested: you pass in a dataset, and it will take care of loading it in batches, see these [lines](https://github.com/autogluon/autogluon/blob/master/timeseries/src/autogluon/timeseries/models/chronos/model.py#L556-L577).
